Research/lva-node-app/public/js/main.js

```diff
@@ -0,0 +1,638 @@
+    return v.toString(16);
+  });
+}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+//configuration page functions
+/** 
+* grab inputs from configuration. Currently set expiration to 1 day. Calls to connect to device and IotHub to make sure your credentials are valid!
+*/
+function sendConfigData() 
+{
+    let payload =
+    {
+        "device-id": document.getElementById("device-id").value,
+        "iothub-connection-string": document.getElementById("iothub-connection-string").value
+    };
+
+    var request = sendRequest(payload, `http://localhost:${PORT}/connectToIotHub`);
+    request.onreadystatechange = function () 
+    {
+        if (request.readyState == 4 && request.status == 200) 
+        {
+            document.getElementById("configuration-output-box").innerHTML = request.response;
+        }
+    }
+}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+//invoking methods/output related methods
+
+/** 
+* show current graph instances in dropdown. Used in output.html when invoking methods 
+* @param {HTMLButtonElement} htmlElement - button user clicks on
+*/
+function instanceMethodDropdowns(htmlElement) 
+{
+  let hoverList = htmlElement.parentElement.getElementsByTagName("ul")[0];
+  let onclickMethod = 'graphEntityModify(this, "", false, false)';
+  hoverList.innerHTML = "";
+  let checkBoxNameList = [];
+  switch(hoverList.id)
+  {
+    case 'instance-list':
+      checkBoxNameList=Object.keys(graphInstances);
+      break;
+    case 'topology-list':
+      checkBoxNameList=Object.keys(graphTopologies);
+      break;
+    case 'topology-set':
+      checkBoxNameList=Object.keys(graphTopologies);
+      onclickMethod='graphSetTopology(this)';
+      break;
+  }
+
+  checkBoxNameList.forEach((item) => 
+  {
+    hoverList.innerHTML += "<li> <button class='btn btn-outline-secondary' data-toggle='dropdown' onclick='"+onclickMethod+"' name='" + htmlElement.getAttribute('name') + "'type='checkbox'>" + item + "</button> </li>";
+  });
+}
+
+/**
+* Create the larger payload object to be sent to the server to invoke a direct method on the lvaEdge module
+* @param {string} methodNameParam - name of method to invoke
+* @param {jsonObject} methodPayload - payload for method
+* @returns {jsonObject} full payload for request
+*/
+function createFullPayload(methodNameParam, methodPayload)
+{
+  let fullPayload = 
+  {
+    methodName: methodNameParam,
+    responseTimeoutInSeconds: 200,
+    Payload: methodPayload  
+  }
+  return fullPayload;
+}
+
+/**
+* see sample graph topologies here: https://github.com/Azure/live-video-analytics/tree/master/MediaGraph/topologies
+* @param {string | HTMLButtonElement} htmlElement - either a string graph name or the HTML button containing the name of the graph
```

Comment:
this semantic feels a little weird... would it be possible to have the upper layer send the graphname instead?

Reply:
Found a way to make your suggestion work!